numba/cuda/cudadrv/devicearray.py

```diff
@@ -119,14 +119,20 @@ def __cuda_array_interface__(self):
         else:
             ptr = 0
 
-        if array_core(self).flags['C_CONTIGUOUS']:
-            strides = None
-        else:
-            strides = tuple(self.strides)
+        # Check if `self` is contiguous. Notice, we do not use `array_core()`
+        # since it calls `require_context()`, which can be quite expensive
+        is_contiguous = True
+        size = self.dtype.itemsize
+        for shape, stride in zip(reversed(self.shape), reversed(self.strides)):
+            if shape > 1:
+                if size != stride:
+                    is_contiguous = False
+                    break
+                size *= shape
```

Comment:
I wonder if this check could be made as a standalone helper function so that it can be reused in other places where `require_context()` is also unwanted?

Reply:
Agree, that is a good idea. I have implemented as a help function here: https://github.com/numba/numba/pull/5189/commits/090997099c510be17f4f7c1f7d883ec5dbe46fef

I could also add it as a `DeviceNDArrayBase.is_contiguous()` method instead?